### PR TITLE
Split MajorMinor into two fields

### PR DESCRIPTION
**Pull request by dims**
this reflects what is in the docker/docker/pkg/mount API. makes it
easier to port additional code to out API.

### CODE REVIEW

**Comment by dims** [comment]
Example of porting cadvisor where this was a bit hard - https://github.com/google/cadvisor/pull/2383/files

**Comment by dims** [comment]
Bonus: it does not affect k/k in any way as we don't use the field - https://cs.k8s.io/?q=MajorMinor&i=nope&files=&repos=kubernetes/kubernetes

File: mount/mount_helper_unix.go [hunk 1]
```diff
@@ -106,10 +110,18 @@ func ParseMountInfo(filename string) ([]MountInfo, error) {
 		if err != nil {
 			return nil, err
 		}
+		mm := strings.Split(fields[2], ":")
+		if len(mm) != 2 {
+			return nil, fmt.Errorf("parsing '%s' failed: unexpected minor:major pair %s", line, mm)
+		}
+		major, _ := strconv.Atoi(mm[0])
```
Comment (thockin) [inline mount/mount_helper_unix.go]: Why not actually handle errors?

Comment (dims) [reply mount/mount_helper_unix.go]: Mirroring existing behavior that we may be depending on - https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/docker/docker/pkg/mount/mountinfo_linux.go#L52-L53

i haven't started at enough of these st_dev(s) to say if that's a good idea or a bad idea :)

Comment (dims) [reply mount/mount_helper_unix.go]: Also currently we don't throw an error if the string has bad contents ...

**Comment by thockin** [comment]
Past misbehavior does not forgive perpetuating it :)

**Comment by dims** [comment]
+1 @thockin fixing it now.

**Comment by dims** [comment]
DONE @thockin PTAL. Added a bunch of tests for the new behavior.

**Comment by codenrhoden** [comment]
LGTM. Nice work with the added tests. Not adding `/lgtm` since I don't want this to merge before @thockin can take another look.

File: mount/mount_helper_unix.go [hunk 2]
```diff
@@ -106,10 +110,24 @@ func ParseMountInfo(filename string) ([]MountInfo, error) {
 		if err != nil {
 			return nil, err
 		}
+		mm := strings.Split(fields[2], ":")
+		if len(mm) != 2 {
+			return nil, fmt.Errorf("parsing '%s' failed: unexpected minor:major pair %s", line, mm)
+		}
+		major, err := strconv.Atoi(mm[0])
+		if err != nil {
+			return nil, fmt.Errorf("parsing '%s' failed: unable to parse major device id", mm[0])
```
Comment (thockin) [inline mount/mount_helper_unix.go]: nit: can you print the error, too?  In the unlikely event this fires....

Comment (dims) [reply mount/mount_helper_unix.go]: Done!

**Comment by dims** [comment]
/hold cancel

thanks @thockin

**Pull request merged**